app/app.go

```diff
@@ -173,22 +169,35 @@ func New(dir string, options ...Option) (app *App, err error) {
 			nodeBindAddress = fmt.Sprintf("@snap.%s.dqlite-%d", snapInstanceName, info.ID)
 		}
 
-		nodeDial = makeNodeDialFunc(o.TLS.Dial)
-	} else {
+	} else if o.Conn == nil {
 		nodeBindAddress = info.Address
-		nodeDial = client.DefaultDialFunc
 	}
+
+	// Start the local dqlite engine.
 	node, err := dqlite.New(
 		info.ID, info.Address, dir,
 		dqlite.WithBindAddress(nodeBindAddress),
-		dqlite.WithDialFunc(nodeDial),
 		dqlite.WithFailureDomain(o.FailureDomain),
 		dqlite.WithNetworkLatency(o.NetworkLatency),
 		dqlite.WithSnapshotParams(o.SnapshotParams),
 	)
 	if err != nil {
 		return nil, fmt.Errorf("create node: %w", err)
 	}
+
+	// Set the appropriate dqlite dial function before starting.
+	if o.Conn != nil {
+		err = node.SetExternalDialFunc(o.Conn.dialFunc)
```

Comment:
Was there no way to keep using `WithDialFunc` ?

Reply:
We want both the Node context (parent) and 5-second timed dial context (child) to be exposed to the dial function. We can't pass in the former if the dial func has already been defined, so either we can't use `WithDialFunc` or we would have to add a `parentCtx` parameter to `client.DialFunc`.